### PR TITLE
Add ConsoleRunnerTest for writability check

### PR DESCRIPTION
**Pull request by rvignesh89**
@CharliePoole I wrote a simple test to assert PR #119 

Let me know if this is what you were aiming for. If so then I'll refactor the Mocks using NSubstitue to simplify the code. And I'll also give it a good name 😆

### CODE REVIEW

File: src/NUnitConsole/nunit3-console.tests/CosoleRunnerTests.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using NUnit.Engine.Extensibility;
+using NUnit.Engine.Services;
+using NUnit.Framework;
+
+namespace NUnit.ConsoleRunner.Tests
+{
+    class CosoleRunnerTests
+    {
+        [Test]
+        public void XXX()
```
Comment (rprouse) [inline src/NUnitConsole/nunit3-console.tests/CosoleRunnerTests.cs]: Could you rename this to something more meaningful? Maybe `ThrowsNUnitEngineExceptionWhenTestResultsAreNotWriteable`?

File: src/NUnitConsole/nunit3-console.tests/CosoleRunnerTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            var testEngine = new TestEngine();
+            testEngine.Services.Add(new TestResultService());
+            var consoleRunner = new ConsoleRunner(testEngine, new ConsoleOptions("mock-assembly.dll", "--work=c:\\windows\\"), new ColorConsoleWriter());
+            Assert.Throws<NUnitEngineException>(() => { consoleRunner.Execute(); });
```
Comment (rprouse) [inline src/NUnitConsole/nunit3-console.tests/CosoleRunnerTests.cs]: I would like to check the message in the exception. You can check that the message contains a substring.

**Comment by rvignesh89** [comment]
@rprouse I've made the changes as requested. Although, I couldn't use NSubstitute in my tests as previously planned. This was because the tests are still in .NET 2.0. Any plans to upgrade?

**Comment by rprouse** [comment]
We don't plan to upgrade the .NET version any time soon. A feature of NUnit is that it works on older frameworks, and if you want to run your tests under .NET 2.0, the runner needs to be compiled to the lowest version supported, 2.0.

**Comment by rprouse** [comment]
Looks good, thanks for the help. 👍

**Comment by rprouse** [comment]
@nunit/core-team this needs a second review and merge.

**Comment by ChrisMaddock** [comment]
Looks good to me!

**Pull request merged**